deploy/operator/before_upgrade.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+export SERVICE_IMAGE=""
+export INDEX_IMAGE="quay.io/ocpmetal/assisted-service-index:ocm-2.3"
```

Comment:
Where do we set the `INDEX_IMAGE` to be the newly built operator? I didn't see another value in `after_upgrade.sh`

Reply:
There are 3 phases in the upgrading job, before the upgrade, upgrade and after the upgrade. So, when `baremetalds-assisted-operator-upgrade-ref.yaml` calls  `baremetalds-assisted-operator-upgrade-commands.sh`, I am setting the latest index image and patching the catalogsource with that image.
https://github.com/openshift/release/pull/21539/files#diff-27e52a9f8e18f6f4a309185c25377a2839112270b55e25429bca81156182bd51R35

Reply:
Ah right it is more clear seeing it in code than talking through it. Okay so we will need to ability to override these exports therefore we will need to pass in the variables like we do in other places like:

`export INDEX_IMAGE=$INDEX_IMAGE-:quay.io/ocpmetal/assisted-service-index:ocm-2.3`

But for all of the exports to be safe